### ISSUE TITLE
when reference many bundles (^1-10), use worksheet.get_bundle_uuids

### ISSUE THREAD

**Issue by percyliang** [codalab/codalab-worksheets#66]
Currently, in bundle_cli.py, we're getting one bundle at a time, which is really slow.

For example, in do_rm_command(), we are calling
bundle_uuids = [worksheet_util.get_bundle_uuid(client, worksheet_uuid, bundle_spec) for bundle_spec in args.bundle_spec]
This should be replaced with one call.
Most commands (e.g., kill, add, perm) should be changed.

**Comment by kashizui** [comment]
Taking this on!

**Comment by kashizui** [comment]
Are there any preexisting patterns in our code for batch calls like this? Or should I just give a shot at designing one?

**Comment by percyliang** [comment]
You'll have to write a batch version:
worksheet_util.get_bundle_uuids and local_bundle_client.get_bundle_uuids.
Then have the get_bundle_uuid function call the get_bundle_uuids version for backward compatibility.

**Comment by kashizui** [comment]
Should there also be a batch version of canonicalize.get_bundle_uuid then?

And am I right in assuming that the major bottleneck is probably the multiple calls to model.get_bundle_uuids? In which case that should probably also be refactored to handle batch queries?

**Comment by percyliang** [comment]
You're right that we can make this even more efficient by batching calls to
model, but that would be more involved, and is probably not high priority
right now, so let's just be content with batching calls to the client.

**Comment by kashizui** [comment]
Got it.

Since `worksheet_util.get_bundle_uuid` allows the bundle_spec to explicitly specify a different `worksheet_uuid`, should the batch version of `local_bundle_client.get_bundle_uuids` be able to handle different `worksheet_uuid`s for each `bundle_spec`, or should the calls to the client be batched according to the worksheet (each different worksheet incurs a new call to the client)?

tl;dr should `local_bundle_client.get_bundle_uuids` still only take a single `worksheet_uuid` for a list of `bundle_specs`?

**Comment by percyliang** [comment]
Good question - Let's keep it simple and say all queries have to be with
respect to the same worksheet.

**Comment by kashizui** [comment]
How should we handle bundle_specs in the batch with the "worksheet_spec/bundle_spec" format then? Should they somehow be split up into different batches?

Sorry for all the questions, hopefully I'll get hang soon of what our design priorities are for the project!
